src/game_inventory.cpp

```diff
@@ -482,6 +511,52 @@ class comestible_inventory_preset : public inventory_selector_preset
             return dummy;
         }
 
+        const std::string get_time_left_rounded( const item_location &loc ) {
+            const item *item = loc.get_item();
+            const double relative_rot = item->is_food_container() ? item->contents.front().get_relative_rot() :
+                                        item->get_relative_rot();
+            const time_duration shelf_life = get_edible_comestible( loc ).spoils;
+            time_duration time_left = shelf_life - shelf_life * relative_rot;
+
+            const float days_left = to_days<float>( time_left );
+            if( days_left <= 0.25 ) {
+                if( item->is_going_bad() ) {
+                    return _( "soon!" );
+                } else {
+                    time_left = time_duration::from_hours( 6 );
+                }
+            } else if( days_left <= 7 ) {
+                time_left = time_duration::from_days( ceilf( days_left ) );
+            } else {
+                time_left = time_duration::from_days( ceilf( days_left / 7 ) * 7 );
+            }
+            if( time_left > get_edible_comestible( loc ).spoils ) {
+                time_left = get_edible_comestible( loc ).spoils;
+            }
+            return to_string_clipped( time_left );
```

Comment:
Can't it be replaced with `to_string_approx`? This could probably make all the conversions above unneeded:

```c++
if( time_left <= 6_hours && item->is_going_bad() ) {
    return _( "soon!" );
}

return to_string_approx( time_left );
```

Also, `std::ceil` is preferred to `ceilf`.

Reply:
I'm not sure why the original code didn't use `to_string_approx`, but I just found out about it and I like it, so I'm going to use it.

![To String Approx](https://i.imgur.com/h35LHDw.png)

Reply:
I'm not sure why the original code didn't use `to_string_approx`, but I just found out about it and I like it, so I'm going to use it.

![To String Approx](https://imgur.com/h35LHDw)